aspnet/security/authorization/secure-data.rst

```diff
@@ -0,0 +1,360 @@
+
+.. code-block:: none
+
+  dotnet user-secrets set SeedUserPW <PW>
+
+Update ``Configure`` to use the test password:
+
+.. literalinclude::  secure-data/samples/final/Startup.cs
+  :language: c#
+  :start-after: #region snippet_SeedUserPW
+  :end-before: #endregion
+  :dedent: 12
+
+Add the test accounts user ID to the seed data. Only one contact is shown, add the user ID to all contacts:
+
+.. literalinclude::  secure-data/samples/final/Data/SeedData.cs
+  :language: c#
+  :start-after: #region snippet_Contact
+  :end-before: #endregion
+  :dedent: 12
+  :emphasize-lines: 9
+
+Delete all the records in the ``Contact`` table and restart the app to seed the database. If you're using Visual Studio you might need to stop IIS Express to force the seed initializer to run.
+
+Resource based authorization
+-----------------------------
+
+- Create an *Authorization* folder for the handlers and classes we will create to implement authorization.
+- Create a *ContactIsOwnerAuthorizationHandler* class we can invoke to verify the user acting on the resource owns the resource. Create this in the *Authorization* folder.
+
+.. literalinclude:: secure-data/samples/final/Authorization/ContactIsOwnerAuthorizationHandler.cs
+  :language: c#
+
+The ``ContactIsOwnerAuthorizationHandler`` calls ``context.Succeed`` if the current authenticated user is the contact owner. We allow contact owners to perform any operation on their own data, so we don't need to check the operation passed in the requirement parameter,
+
+Services using Entity Framework Core must be registered for :ref:`dependency injection <fundamentals-dependency-injection>` using :dn:method:`~Microsoft.Extensions.DependencyInjection.ServiceCollectionServiceExtensions.AddScoped`. The ``ContactIsOwnerAuthorizationHandler`` uses ASP.NET Core Identity, which is built on Entity Framework Core. Register the ``ContactIsOwner`` handler with the service collection so it will be available to the ``ContactsController`` through :ref:`dependency injection <fundamentals-dependency-injection>`. Add the following code to the end of ``ConfigureServices``:
+
+.. literalinclude::  secure-data/samples/final/Startup.cs
+  :language: c#
+  :start-after: snippet_AddScoped
+  :end-before: #endregion
+  :dedent: 12
+
+Update the ``ContactsController`` constructor to resolve the ``IAuthorizationService`` service so we'll have access to our authorization handlers we have registered. While we're at it we'll also get the ``Identity`` ``UserManager`` service:
+
+.. literalinclude:: secure-data/samples/final/Controllers/ContactsController.cs
+  :language: c#
+  :start-after: snippet_ContactsController
+  :end-before: #endregion
+  :dedent: 4
+  :emphasize-lines: 4,5,9,10,13,14
+
+Add the ``ContactOperationsRequirements`` class to the *Authorization* folder to contain the requirements our app supports:
+
+.. literalinclude:: secure-data/samples/final/Authorization/ContactOperationsRequirements.cs
```

Comment:
Per blowdart: (from PR #1734)
This appears to be mixed up operations and permissions. For example you've defined the name of the create operation as canCreate, which is the actual permission. This isn't what I'd expect, I'd expect the operation name to be create, and then you have a mapping table to map the permission role or claim to the actual operation name inside the handler, or in a helper.